keps/sig-storage/2324-volume-snapshot-namespace-transfer/README.md

```diff
@@ -0,0 +1,538 @@
+
+We must allow the user to specify new secrets, in the event that they do not exist in the target namespace.
+These can be provided in the `StorageTransferApproval`, as outlined above.
+
+- What should be done when the PVC is currently in use?
+
+The current approach of using a Finalizer and deleting the source PVC should address this, as it will 
+wait for all running Pods to stop using the PVC before the transfer continues. 
+
+This process can take some time, and we need to ensure users who request a transfer have the ability
+to check the status of the transfer.
+
+- Can a transfer be performed when the PVC is having an operation performed on it?
+
+This issue is less of a concern for `VolumeSnapshots`, as operations are typically not performed on the
+snapshot, but for PVCs it is possible the user could be resizing or snapshotting the PVC during
+the transfer request.
+
+The current approach of using a Finalizer and deleting the source PVC should address this concern.
+For instance, the external-resizer can resume the operation on a PVC after the transfer is completed.
+We should ensure that other sidecars can continue once the transfer is completed as well.
+
+- As long as we reference the existing VolumeSnapshotContent, there should be no issues with encrypted VolumeSnapshotContents. The secret reference is stored on the VolumeSnapshotContent and VolumeSnapshotClass, not the VolumeSnapshot itself. If we take a different approach, such as re-creating the VolumeSnapshotContent, then we will need to ensure these credentials are propagated to the new content.
+- The VolumeSnapshot must be `readyToUse` before the transfer begins.
+- Any owners of the destination namespace must accept the transfer. By using the StorageTransfer API, this ensures an owner in the destination namespace is aware of the transfer.
+
+## Design Details
+
+This continues the discussion from [API for PVC Namespace Transfer](https://github.com/kubernetes/enhancements/pull/2110/) 
+in regards to the StorageTransfer API.
+
+A `StorageTransferRequest` resource will be created in the target namespace:
+
+```yaml
+apiVersion: storage.k8s.io/v1alpha1
+kind: StorageTransferRequest
+metadata:
+  name: kAnUOwpzso
+  namespace: target-namespace
+spec:
+  approvalName: xyji6OwynW
+  targetName: bar
+```
+
+`StorageTransferApproval` resource will be created in the source namespace:
+
+```yaml
+apiVersion: storage.k8s.io/v1alpha1
+kind: StorageTransferApproval
+metadata:
+  name: xyji6OwynW
+  namespace: source-namespace
+spec:
+  source:
+    name: foo
+    kind: VolumeSnapshot
+  secrets:
+    - name: secret-name
+      namespace: secret-namespace
+      type: secret-type # such as `csi.storage.k8s.io/provisioner-secret-name`
+  targetNamespace: target-namespace
+  requestName: kAnUOwpzso
+```
+
+NOTE: The name of the `StorageTransferApproval` and `StorageTransferRequest` should not be easy to guess,
```

Comment:
Should this matter?  The design should not allow things like "guessing" of the transfer name to allow unintended transfer. Obscurity isn't a great defense, but in the model if we're doing a grant/accept model there shouldn't be much room for abuse, especially if you throw in the possibility of requiring a token .

Reply:
I've included a token here and removed this note.